feat(python): Add Typer integration docs (#12174)
getsentry/sentry-docs: docs/platforms/python/integrations/typer/index.mdx

---
title: Typer
description: Learn how to use Sentry to capture Typer exceptions.
---

> The `TyperIntegration` captures exceptions raised when using Typer CLI applications.

Reviewer: ```suggestion
The `TyperIntegration` captures exceptions raised when using [Typer CLI](https://typer.tiangolo.com/) applications.
```

## Install

Install Typer and the Sentry Python SDK.

```bash
pip install --upgrade "sentry-sdk" typer
```

## Configure

To enable the `TyperIntegration`, add it to the `integrations` list of your `sentry_sdk.init`. 

```python
import sentry_sdk
from sentry_sdk.integrations.typer import TyperIntegration

sentry_sdk.init(
    dsn="___PUBLIC_DSN___",
    integrations=[TyperIntegration()],
)
```

## Verify

Create a small CLI application:

```python
import typer
import sentry_sdk
from sentry_sdk.integrations.typer import TyperIntegration

sentry_sdk.init(
    dsn="___PUBLIC_DSN___",
    integrations=[TyperIntegration()],
)

def main():
    1 / 0

if __name__ == "__main__":
    typer.run(main)
```

When you run this, Sentry will capture the `ZeroDivisionError` from the `main()`

> function and you will be able to see it on [sentry.io](https://sentry.io).

Reviewer: ```suggestion
function and you'll be able to see it on [sentry.io](https://sentry.io).
```
